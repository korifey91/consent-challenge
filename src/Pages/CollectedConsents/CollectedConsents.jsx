import { useCallback, useState } from 'react';

import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TablePagination from '@material-ui/core/TablePagination';
import TableRow from '@material-ui/core/TableRow';

import styles from './CollectedConsents.module.scss';

const columns = [
  { id: 'name', label: 'Name', minWidth: 100 },
  { id: 'email', label: 'Email', minWidth: 170 },
  { id: 'consent', label: 'Consent given for', minWidth: 250 },
];

const rows = [
  {
    id: 1,
    name: 'User Friendly',
    email: 'user@friendly.com',
    consents: ['Receive newsletter'],
  },
  {
    id: 2,
    name: 'User2 Friendly',
    email: 'user2@friendly.com',
    consents: ['Receive newsletter', 'Be shown targeted ads'],
  },
  {
    id: 3,
    name: 'User3 Friendly',
    email: 'user3@friendly.com',
    consents: [
      'Receive newsletter',
      'Be shown targeted ads',
      'Contribute to anonymous visit statistics',
    ],
  },
];

const rowsPerPage = 2;

function CollectedConsents() {
  const [page, setPage] = useState(0);
  const handleChangePage = useCallback((event, newPage) => {
    setPage(newPage);
  }, []);

  return (
    <div className={styles.table}>
      <TableContainer>
        <Table size="medium">
          <TableHead>
            <TableRow>
              {columns.map((column) => (
                <TableCell
                  key={column.id}
                  align={column.align}
                  style={{ minWidth: column.minWidth }}
                >
                  {column.label}
                </TableCell>
              ))}
            </TableRow>
          </TableHead>
          <TableBody>
            {rows
              .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
              .map((row) => (
                <TableRow hover key={row.id}>
                  <TableCell component="th" scope="row">
                    {row.name}
                  </TableCell>
                  <TableCell>{row.email}</TableCell>
                  <TableCell>{row.consents.join(', ')}</TableCell>
                </TableRow>
              ))}
          </TableBody>
        </Table>
      </TableContainer>
      <TablePagination
        component="div"
        count={rows.length}
        rowsPerPageOptions={[2]}
        rowsPerPage={rowsPerPage}
        page={page}
        onChangePage={handleChangePage}
      />
    </div>
  );
}

export default CollectedConsents;
