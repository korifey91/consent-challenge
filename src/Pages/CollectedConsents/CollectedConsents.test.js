import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { unmountComponentAtNode } from 'react-dom';

import configureStore from 'redux-mock-store';
import { render } from '@testing-library/react';

import CollectedConsents from './CollectedConsents';

const mockStore = configureStore([]);

let container = null;
let store = null;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);

  store = mockStore({
    consents: {
      data: [],
      meta: {
        totalCount: null,
        totalPages: null,
        currentPage: null,
      },
      error: null,
    },
  });
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

test('renders CollectedConsents correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <CollectedConsents />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders collected consents from store', () => {
  store = mockStore({
    consents: {
      data: [
        {
          id: 1,
          name: 'User Friendly',
          email: 'user@friendly.com',
          consents: ['Receive newsletter'],
        },
      ],
      meta: {
        totalCount: null,
        totalPages: null,
        currentPage: null,
      },
      error: null,
    },
  });

  const { container, getByText } = render(
    <Provider store={store}>
      <CollectedConsents />
    </Provider>
  );

  expect(getByText('User Friendly')).toBeInTheDocument();
});
