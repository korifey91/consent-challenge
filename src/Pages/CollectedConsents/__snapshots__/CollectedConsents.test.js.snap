// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CollectedConsents correctly 1`] = `
<div
  className="table"
>
  <div
    className="MuiTableContainer-root"
  >
    <table
      className="MuiTable-root"
      role={null}
    >
      <thead
        className="MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head"
            scope="col"
            style={
              Object {
                "minWidth": 100,
              }
            }
          >
            Name
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head"
            scope="col"
            style={
              Object {
                "minWidth": 170,
              }
            }
          >
            Email
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head"
            scope="col"
            style={
              Object {
                "minWidth": 250,
              }
            }
          >
            Consent given for
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root"
        role={null}
      />
    </table>
  </div>
  <div
    className="MuiTablePagination-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
    >
      <div
        className="MuiTablePagination-spacer"
      />
      <p
        className="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
      >
        0-0 of 0
      </p>
      <div
        className="MuiTablePagination-actions"
      >
        <button
          aria-label="Previous page"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          title="Previous page"
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
              />
            </svg>
          </span>
        </button>
        <button
          aria-label="Next page"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          title="Next page"
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
